pkg/util/system/system_utils.go

```diff
@@ -17,11 +17,20 @@ limitations under the License.
 package system
 
 import (
-	"regexp"
+	"strings"
 )
 
 // TODO: find a better way of figuring out if given node is a registered master.
 func IsMasterNode(nodeName string) bool {
-	r := regexp.MustCompile("master(-...)?$")
```

Comment:
That's not true. Only 10-15% was coming from compile phase. The rest was from processing.